api_loader.go

```diff
@@ -521,15 +523,15 @@ func (d *DummyProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 func loadGlobalApps() {
 	// we need to make a full copy of the slice, as loadApps will
 	// use in-place to sort the apis.
-	apisMu.RLock()
-	specs := make([]*APISpec, len(apiSpecs))
-	copy(specs, apiSpecs)
-	apisMu.RUnlock()
-	loadApps(specs, mainRouter)
```

Comment:
Perhaps undo these changes, as well as `getApiSpecs`. There is no change of behaviour that I see, and you're adding more code.